main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/csv"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type Pokemon struct {
+	Id      int    `json:"id"`
+	Name    string `json:"name"`
+	TypeOne string `json:"typeOne"`
+}
+
+func main() {
+
+	pokemons := initializePokemonData()
+
+	r := mux.NewRouter()
+	r.HandleFunc("/pokemon/{id}", func(rw http.ResponseWriter, r *http.Request) {
+		HandlePokemonById(rw, r, pokemons)
+	})
+	http.ListenAndServe(":8080", r)
+}
+
+func initializePokemonData() []Pokemon {
+	csvfile, err := os.Open("pokemons.csv")
+	if err != nil {
+		log.Fatalln("Couldn't open the csv file", err)
+	}
+
+	csvR := csv.NewReader(csvfile)
+
+	records, recordsErr := csvR.ReadAll()
+
+	if recordsErr != nil {
+		log.Fatalln("No records")
```

Comment:
Again, you would like to return and end the flow if something happened, we wouldn't like to rely on log.Fatal to stop the flow., specially if we're moving this to another package